### DISCUSSION TITLE
LilyGo T-Display ESP32-S3 strange warning in pio: right shift count >= width of type [-Wshift-count-overflow]

### DISCUSSION THREAD

**Discussion by pdurys** [Bodmer/TFT_eSPI#2519] [Compatible displays and setup files]
Hi there,

This is just warning. Application works as expected. I am trying to nail compiler why is shouting on me.
In standard Setup206_LilyGo_T_Display_S3.h file i have warnings "right shift count >= width of type [-Wshift-count-overflow]" in lines TFT_D0 - TFT_D7

```
#define TFT_WR 8
#define TFT_RD 9

#define TFT_D0 39
#define TFT_D1 40
#define TFT_D2 41
#define TFT_D3 42
#define TFT_D4 45
#define TFT_D5 46
#define TFT_D6 47
#define TFT_D7 48

#define TFT_BL 38
```

Does anybody knows why is that?

As said it is not a problem at all. I just do not understand why preprocessor / complier shouts that error on several lines not bother about others.

**Comment by Bodmer** [comment]
I think this is sorted in the github master. It will only cause a problem for TFT reads.

**Reply by pdurys** [reply]
OK, but why?
For me #define is just find&replace clause to substitute eg. TFT_D2 for 41
Why compiler complains this?
Is that mean that somewhere in code TFT_D2 is declared in such way that integer 41 cause issues?
